src/main/java/com/cisco/amp/flink/SecTweet.java

```diff
@@ -46,82 +47,117 @@
-        esSinkBuilder.setBulkFlushMaxActions(1);
-        tokenCountDataStream.addSink(esSinkBuilder.build());
-    }
-
     void buildJobGraph(StreamExecutionEnvironment env, ParameterTool params) {
         DataStreamSource<String> streamSource;
+
         if (params.has(PARAM_FILE_KEY)) {
+            // File source
             streamSource = env.readTextFile(params.get(PARAM_FILE_KEY));
         } else {
+            // Streaming source
             if (!(params.has(TwitterSource.CONSUMER_KEY) && params.has(TwitterSource.CONSUMER_SECRET)
                 && params.has(TwitterSource.TOKEN) && params.has(TwitterSource.TOKEN_SECRET))) {
                 System.out.println("Usage: --twitter-source.consumerKey <key> --twitter-source.consumerSecret <secret> "
                     + "twitter-source.token <token> --twitter-source.tokenSecret <tokenSecret>");
                 return;
             }
-            // Get input data
+            List<Long> twitterIds = new ArrayList<>();
+            for (String id : params.get(PARAM_TWITTER_SOURCE_IDS_KEY).split(",")) {
+                twitterIds.add(Long.parseLong(id.trim()));
+            }
+            List<String> twitterTerms = new ArrayList<>();
+            for (String term : params.get(PARAM_TWITTER_SOURCE_TERMS_KEY).split(",")) {
+                twitterTerms.add(term.trim());
+            }
+
             TwitterSource twitterSource = new TwitterSource(params.getProperties());
-            twitterSource.setCustomEndpointInitializer(new SecurityEndpointInitializer());
+            twitterSource.setCustomEndpointInitializer(new SecurityEndpointInitializer(twitterIds, twitterTerms));
             streamSource = env.addSource(twitterSource);
         }
 
+        // extract tweet text
         DataStream<Tweet> tweets = streamSource
             .flatMap(new ExtractTweet())
             .assignTimestampsAndWatermarks(new TweetTimestampExtractor(Time.seconds(MAX_LATENESS_SECONDS)));
 
-        DataStream<TokenCount> tokens = tweets
-                .flatMap(new TweetJsonMap())
-                .assignTimestampsAndWatermarks(new AscendingTimestampExtractor<TokenCount>() {
-                    @Override
-                    public long extractAscendingTimestamp(TokenCount element) { return element.getTimestamp(); }
-                });
-        DataStream<TokenCount> tokenCountDataStream = countTokens(tokens, DEFAULT_RATE_INTERVAL);
-        DataStream<TokenTrend> trendsDataStream = getTrends(tokenCountDataStream, DEFAULT_TREND_WINDOW_SIZE, DEFAULT_TREND_WINDOW_SLIDE);
+        // extract tokens
+        DataStream<TokenCount> tokens = tweets.flatMap(new TweetJsonMap());
 
-        trendsDataStream.print();
+        // Get the token rate
+        DataStream<TokenCount> tokenCountDataStream = countTokens(tokens, DEFAULT_RATE_INTERVAL);
 
-        if (params.has(WRITE_ES_KEY)) {
+        // Optional: Write to ES for visualization
```

Comment:
Writing to ES is not purely for visualization tho, this could be misleading.

Reply:
Updated